### PR TITLE
Updated BearerAuthentication to create users

### PR DESCRIPTION
**Pull request by clintonb**
- BearerAuthentication can, optionally, create users
- All settings defined in a package-specific dict
- Documentation!

ECOM-3895

### CODE REVIEW

**Comment by clintonb** [comment]
Docs available at https://edx-django-rest-framework-extensions.readthedocs.org/en/clintonb-oauth-user-info/.

File: edx_rest_framework_extensions/authentication.py [hunk 1]
```diff
@@ -54,39 +57,88 @@ def authenticate_credentials(self, token):
...
+            ('username', 'preferred_username'),
+            ('email', 'email'),
+            ('last_name', 'family_name'),
+            ('first_name', 'given_name'),
+        )
+
+        for dest, source in mapping:
+            user_info[dest] = response[source]
+
+        return user_info
```
Comment (peter-fogg) [inline edx_rest_framework_extensions/authentication.py]: `return {dest: response[source] for dest, source in mapping}`

File: docs/settings.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+These settings are used by the ``BearerAuthentication`` class.
+
+``OAUTH2_USER_INFO_URL``
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Default: ``None``
+
+URL of an endpoint on the OAuth2 provider where ``BearerAuthentication`` can retrieve details about the user associated
+with the provided access token. This endpoint should return a JSON object with user details and ``HTTP 200`` if, and
```
Comment (peter-fogg) [inline docs/settings.rst]: Could document the format of user details.

File: edx_rest_framework_extensions/authentication.py (hunk 1 above)
Comment (clintonb) [inline edx_rest_framework_extensions/authentication.py]: Updated.

File: docs/settings.rst (hunk 2 above)
Comment (clintonb) [inline docs/settings.rst]: Updated.

**Comment by clintonb** [comment]
@peter-fogg any additional feedback?

**Comment by peter-fogg** [comment]
@clintonb Nope. :+1:

**Comment by clintonb** [comment]
Thanks.

**Pull request merged**